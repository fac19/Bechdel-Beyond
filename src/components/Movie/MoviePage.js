import React from 'react';
import { Link } from 'react-router-dom';

import { makeStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import Grid from '@material-ui/core/Grid';
import { Button } from '@material-ui/core';

import GetMovie from './Movie/GetMovie';
import GetMovieDetails from './MovieDetails/GetMovieDetails';
import getData from '../../utils/getData';
import {
	GetMovieStatsCast,
	GetMovieStatsCrew,
} from './MovieStats/GetMovieStats';

const useStyles = makeStyles((theme) => ({
	root: {
		flexGrow: 1,
	},
	paper: {
		padding: theme.spacing(2),
		textAlign: 'center',
		color: theme.palette.text.secondary,
	},
}));

export default function MoviePage({ match: { params } }) {
	const classes = useStyles();
	const [movieData, setMovieData] = React.useState('');
	const [reviewData, setReviewData] = React.useState('');
	const [statsCrewData, setCrewStatsData] = React.useState('');
	const [statsCastData, setCastStatsData] = React.useState('');

	React.useEffect(() => {
		console.log(params);
		getData(`/film/${params.title}`).then((data) => setMovieData(data));
		getData(`/film/${params.title}/reviews`).then((data) =>
			setReviewData(data),
		);
		getData(`/film/${params.title}/crew`).then((data) =>
			setCrewStatsData(data),
		);
		getData(`/film/${params.title}/cast`).then((data) =>
			setCastStatsData(data),
		);
	}, []);

	if (!movieData || !reviewData || !statsCrewData || !statsCastData)
		return <p>Loading</p>;

	return (
		<div className={classes.root}>
			<Grid container spacing={3}>
				<Grid item xs={12}>
					<Paper className={classes.paper}>
						<GetMovie movieData={movieData} reviewData={reviewData} />
					</Paper>
				</Grid>
				<Grid item xs={12}>
					<Paper className={classes.paper}>
						<GetMovieDetails
							movieData={movieData}
							statsCrewData={statsCrewData}
						/>
						<Button
							variant="outlined"
							component={Link}
							to={`/create-review/${params.title}`}
						>
							Write Review
						</Button>
					</Paper>
				</Grid>
				<Grid item xs={6}>
					<Paper className={classes.paper}>
						<GetMovieStatsCast statsCastData={statsCastData} />
					</Paper>
				</Grid>
				<Grid item xs={6}>
					<Paper className={classes.paper}>
						<GetMovieStatsCrew statsCrewData={statsCrewData} />
					</Paper>
				</Grid>
			</Grid>
		</div>
	);
}
