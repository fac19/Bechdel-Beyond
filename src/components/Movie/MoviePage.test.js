import React from 'react';
import renderer from 'react-test-renderer';

import MoviePage from './MoviePage';

test('Check movie page tests are running', () => {
	expect(true).toBeTruthy();
});

it('Movie page renders correctly', () => {
	const params = 'titanic';
	const moviePage = renderer
		.create(<MoviePage match={params}></MoviePage>)
		.toJSON();
	expect(moviePage).toMatchSnapshot();
});
