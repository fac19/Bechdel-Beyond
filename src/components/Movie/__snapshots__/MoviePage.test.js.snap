// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie page renders correctly 1`] = `
<p>
  Loading
</p>
`;
